### PR TITLE
fix: remove image tag only not the image using id

### LINKED ISSUE #2673: Delete Action deletes all tags with the same Image ID
### Bug description

When I have a single image with multiple tags but the same `Image ID` clicking the Delete Action button deletes all the images with the same `Image ID`

image

Note: all the images have the same `Image ID == 4821cabd3279`

Clicking the Delete action for a single image deletes all of them. 

From the CLI if the image has multiple tags `podman image rm <image tag>` only removes the tag 
```
$ podman image rm quay.io/tuckerrc/hello
Untagged: quay.io/tuckerrc/hello:latest
```

If the image is the last tag for the image `podman image rm` deletes the image too
```
$ podman image ls
REPOSITORY            TAG         IMAGE ID      CREATED       SIZE
quay.io/podman/hello  latest      4821cabd3279  20 hours ago  142 kB

$ podman image rm quay.io/podman/hello  
Untagged: quay.io/podman/hello:latest
Deleted: 4821cabd3279b5ac0b557e94002cbcffe92b5249cfd94f31e1bf8be36d9577ad
```

### Operating system

macOS Ventura 13.3.1 (Apple silicon)

### Version

next (development version)

### Steps to reproduce

1. Create or Pull an image
    - `podman pull quay.io/podman/hello`
3. From the CLI (or Podman-Desktop if #2588 is complete) add a new tag
    - `podman image tag <current-tag> <new-tag>`
4. In Podman Desktop click the "Delete Action" button for one of the listed images
5. See both images have been deleted

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

* switches to use name:tag to delete image so only specific tag is deleted not the image itself and all the tags
* run delete commands as a chain to avoid conflicts when deleting different tags of the same image

### Screenshot/screencast of this PR

n/a

### What issues does this PR fix or reference?

Fix #2673

### How to test this PR?

Create different tags for the same image and try to delete them using podman-desktop ui. The selected tag should be deleted, not all the tags of the same image.

### CODE REVIEW

**Review by benoitf — COMMENTED** [review]
I think it's missing test where images have same id but different tags, etc

**Comment by dgolovin** [comment]
@benoitf I will do that, but that is rather an integration test that should be executed with real podman. I can mock it, but that would not be a correct validation, because I can mock it to work, but it does not mean it works with actual podman. 

All new code seems to be covered with existing tests.

File: packages/renderer/src/lib/image/ImageDetails.svelte [hunk 1]
```diff
@@ -43,15 +43,13 @@ onMount(() => {
...
+      const tempImage = imageUtils.getImageInfoUI(matchingImage, base64RepoTag);
+      if (tempImage) {
+        image = tempImage;
+        return;
       }
-    } else if (detailsPage) {
-      // the image has been deleted
-      detailsPage.close();
     }
+    detailsPage.close();
```
Comment (benoitf) [inline packages/renderer/src/lib/image/ImageDetails.svelte]: I think having the comment that image was deleted was helpful and with the else clause
here with the early return I think it's less readable why it does `detailsPage.close();`

File: tests/src/model/pages/images-page.ts [hunk 2]
```diff
@@ -78,12 +78,14 @@ export class ImagesPage extends MainPage {
...
-    await waitUntil(async () => await this.imageExists(name), 3000, 900);
+    const imageExists = () => this.imageExists(name);
+    await waitUntil(imageExists, 3000, 900);
     return true;
   }
 
   async waitForImageDelete(name: string): Promise<boolean> {
-    await waitWhile(async () => await this.imageExists(name), 3000, 900);
+    const imageExists = () => this.imageExists(name);
+    await waitWhile(imageExists, 3000, 900);
```
Comment (benoitf) [inline tests/src/model/pages/images-page.ts]: this looks unrelated to tests, just some personal preferences so it should be extracted to a separate PR

File: tests/src/utility/wait.ts [hunk 3]
```diff
@@ -33,13 +33,12 @@ export async function wait(
...
-      ? `Timeout (${timeout} ms) was reach while waiting for condition (${waitFunction.name}) to become '${String(
-          until,
-        )}'`
-      : errorMessage;
   if (sendError) {
+    const message =
+      errorMessage ||
+      `Timeout (${timeout} ms) was reach while waiting for condition (${waitFunction.name}) to become '${String(
+        until,
+      )}'`;
```
Comment (benoitf) [inline tests/src/utility/wait.ts]: extract from this PR

File: tests/src/utility/wait.ts [hunk 4]
```diff
@@ -53,14 +52,14 @@ export async function wait(
...
   timeout = 3000,
   diff = 500,
   sendError = true,
   message = '',
 ): Promise<void> {
-  await wait(waitFunction, true, timeout, diff, sendError, message);
+  return wait(waitFunction, true, timeout, diff, sendError, message);
 }
 
 /**
```
Comment (benoitf) [inline tests/src/utility/wait.ts]: refactoring should be a separate PR, it's not related to the fix

**Review by benoitf — APPROVED** [review]
nice work 👍

**Review by benoitf — CHANGES_REQUESTED** [review]
so I was testing again and noticed that there is one usecase where we have a regression

I have some leftover images and I was no longer able to remove these images

I've checked with the main branch and it's working so it's related to this PR

it's for images without a tag

for example do

```
podman pull quay.io/podman/hello
podman untag quay.io/podman/hello
```

then you see an image like
image


and trying to remove it leads to 

image


then try to remove this image

**Comment by dgolovin** [comment]
Good catch. I remember I've seen images like this, but did not pay enough attention to it :(.

**Review by benoitf — APPROVED** [review]
now it works as expected 👍 

maybe add a test for the `<none>` usecase as it's not covered by a test, so we might forget about it later
